src/emit/mod.rs

```diff
@@ -510,6 +510,15 @@ pub trait TargetRuntime<'a> {
                         .unwrap()
                         .into_pointer_value();
 
+                    let dest = bin.builder.build_pointer_cast(
+                        v,
+                        bin.module
+                            .get_struct_type("struct.vector")
+                            .unwrap()
+                            .ptr_type(AddressSpace::Generic),
+                        "vector",
+                    );
+
```

Comment:
You've modified emit, but no tests for this modification have been provided.

Reply:
Without this change, the test will fail. So it is tested.

There is a problem in the emit phase where sometimes llvm renames type symbols during merge of llvm modules. Rather than merging everything into one module including the stdlib, we should either let the linker deal with this or not merge at all. I'm hoping we can we get rid of the C version of stdlib at some point and this problem will go away.